File: kmp-cmp/shared/src/commonMain/kotlin/data/dto/Dtos.kt
package com.stackbenchmark.kmpcmp.data.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MovieDto(
    val id: Int,
    val title: String,
    val overview: String,
    @SerialName("poster_path") val posterPath: String? = null,
    @SerialName("backdrop_path") val backdropPath: String? = null,
    @SerialName("release_date") val releaseDate: String = "",
    @SerialName("vote_average") val voteAverage: Double = 0.0,
    @SerialName("vote_count") val voteCount: Int = 0
)

@Serializable
data class GenreDto(
    val id: Int,
    val name: String
)

@Serializable
data class MovieDetailDto(
    val id: Int,
    val title: String,
    val overview: String,
    @SerialName("poster_path") val posterPath: String? = null,
    @SerialName("backdrop_path") val backdropPath: String? = null,
    @SerialName("release_date") val releaseDate: String = "",
    @SerialName("vote_average") val voteAverage: Double = 0.0,
    @SerialName("vote_count") val voteCount: Int = 0,
    val tagline: String? = null,
    val runtime: Int? = null,
    val genres: List<GenreDto> = emptyList()
)

@Serializable
data class PopularMoviesResponseDto(
    val page: Int,
    @SerialName("total_pages") val totalPages: Int,
    val results: List<MovieDto>
)

File: rn-expo/src/data/dtos.ts
export interface MovieDto {
  id: number;
  title: string;
  overview: string;
  poster_path: string | null;
  backdrop_path: string | null;
  release_date: string;
  vote_average: number;
  vote_count: number;
}

export interface GenreDto {
  id: number;
  name: string;
}

export interface MovieDetailDto extends MovieDto {
  tagline: string | null;
  runtime: number | null;
  genres: GenreDto[];
}

export interface PopularMoviesResponseDto {
  page: number;
  total_pages: number;
  results: MovieDto[];
}
